src/Servers/Kestrel/Core/src/Internal/Http3/Http3ControlStream.cs

```diff
@@ -0,0 +1,394 @@
+                {
+                    break;
+                }
+                payload = payload.Slice(consumed);
+
+                var value = VariableLengthIntegerHelper.GetVariableIntFromReadOnlySequence(payload, out consumed, out examinded);
+                if (id == -1)
+                {
+                    break;
+                }
+                payload = payload.Slice(consumed);
+                ProcessSetting(id, value);
+            }
+
+            return default;
+        }
+
+        private void ProcessSetting(long id, long value)
+        {
+            switch (id)
+            {
+                case (long)Http3SettingType.QPACK_MAX_TABLE_CAPACITY:
+                    _http3Connection.ApplyMaxTableCapacity(value);
+                    break;
+                case (long)Http3SettingType.MaxHeaderListSize:
+                    _http3Connection.ApplyMaxHeaderListSize(value);
+                    break;
+                case (long)Http3SettingType.QPACK_BLOCKED_STREAMS:
+                    _http3Connection.ApplyBlockedStream(value);
+                    break;
+                default:
+                    // Ignore all unknown settings.
+                    break;
+            }
+        }
+
+        private ValueTask ProcessGoAwayFrameAsync()
+        {
+            if (!_haveReceivedSettingsFrame)
+            {
+                throw new Http3ConnectionException("HTTP_FRAME_UNEXPECTED");
+            }
+            // Get highest stream id and write that to the response.
+            return default;
+        }
+
+        private ValueTask ProcessCancelPushFrameAsync()
+        {
+            if (!_haveReceivedSettingsFrame)
+            {
+                throw new Http3ConnectionException("HTTP_FRAME_UNEXPECTED");
+            }
+            // This should just noop.
+            return default;
+        }
+
+        private ValueTask ProcessMaxPushIdFrameAsync()
+        {
+            if (!_haveReceivedSettingsFrame)
+            {
+                throw new Http3ConnectionException("HTTP_FRAME_UNEXPECTED");
+            }
+            return default;
+        }
+
+        private ValueTask ProcessUnknownFrameAsync()
+        {
+            if (!_haveReceivedSettingsFrame)
+            {
+                throw new Http3ConnectionException("HTTP_FRAME_UNEXPECTED");
+            }
+            return default;
+        }
+
+        public void StopProcessingNextRequest()
+            => StopProcessingNextRequest(serverInitiated: true);
+
+        public void StopProcessingNextRequest(bool serverInitiated)
+        {
+            var initiator = serverInitiated ? GracefulCloseInitiator.Server : GracefulCloseInitiator.Client;
+
+            if (Interlocked.CompareExchange(ref _gracefulCloseInitiator, initiator, GracefulCloseInitiator.None) == GracefulCloseInitiator.None)
+            {
+                Input.CancelPendingRead();
+            }
+        }
+
+        public void Tick(DateTimeOffset now)
+        {
```

Comment:
Will we need to handle timeouts for these control streams?

Reply:
Possibly. I won't know until I get this implementation more fleshed out.